lib/rules/no-implicit-globals.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @fileoverview Rule to check for implicit global variables and functions.
+ * @author Joshua Peek
+ * @copyright 2015 Joshua Peek. All rights reserved.
+ * See LICENSE file in root directory for full license.
+*/
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    return {
+        "Program": function() {
+            var scope = context.getScope();
+
+            scope.variables.forEach(function(variable) {
```

Comment:
@michaelficarra it feels wrong that `scope.variables` includes `const` and `let` declarations. The spec calls for managing them in separate lists on the environment record. Seems like it would be nice the scoping maintained the same. Maybe `scope.variables` and `scope.declarations` (or something)